src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineHelper.cs

```diff
@@ -60,8 +60,9 @@ JToken ParameterFunction(ITextSnapshot snapshot)
                 parameterFactory: ParameterFunction,
                 cancellationToken: cancellationToken).ConfigureAwait(false))?.Response;
 
-            Contract.ThrowIfNull(requestSnapshot);
-            return response is null ? null : (response, requestSnapshot);
+            // The request snapshot or response can be null if there is no LSP server implementation for
+            // the document symbol request for that language.
+            return requestSnapshot is null || response is null ? null : (response, requestSnapshot);
```

Comment:
I noticed when opening a TS file that the progress bar wasn't showing... that's because for a language without the LSP server implementation we always throw since requestSnapshot will be null (not at all the desired effect). Instead, we should just check whether the requestSnapshot or the response is null.

Reply:
I would break this change out into its own PR since we likely want this to go in as-is regardless of the design of the progress bar

Reply:
Sounds good to me, new PR for this fix: https://github.com/dotnet/roslyn/pull/63031